src/utils/api.js

```diff
@@ -1,24 +1,78 @@
+import ArticleDates from "../utils/getDates";
+
 export default class Api {
-  addArticle = (article) => {
-    return new Promise((resolve) => {
-      resolve({
-        urlToImage: article.urlToImage,
-        title: article.title,
-        description: article.description,
-        source: article.source,
-        publishedAt: article.publishedAt,
-        url: article.url,
-        keyword: article.keyword,
-        id: article.id,
-      });
-    });
+  constructor({ baseUrl }) {
+    this._baseUrl = baseUrl;
+    this._articleDates = new ArticleDates();
+    this._pageSize = 100;
+  }
+
+  getNewsArticles(query) {
+    if (query) {
+      return fetch(`${this._baseUrl}/articles/search/${query}`, {
+        method: "GET",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+      }).then((res) => this._checkReponse(res));
```

Comment:
**COULD BE IMPROVED**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.